sdk/python/tests/dsl/component_tests.py

```diff
@@ -43,6 +47,29 @@ def componentA(a: {'ArtifactA': {'file_type': 'csv'}}, b: Integer() = 12, c: {'A
 
     self.assertEqual(containerOp._metadata, golden_meta)
 
+  def test_component_metadata_standard_type_annotation(self):
+    """Test component decorator metadata."""
+
+    class MockContainerOp:
+      def _set_metadata(self, component_meta):
+        self._metadata = component_meta
+
+    @component
+    def componentA(a: float, b: List[int], c: Optional[str] = None) -> None:
+      return MockContainerOp()
+
+    containerOp = componentA('str_value', '[1,2,3]')
+
+    golden_meta = ComponentSpec(name='ComponentA', inputs=[], outputs=None)
+    golden_meta.inputs.append(InputSpec(name='a', type='Float'))
+    golden_meta.inputs.append(
+        InputSpec(name='b', type='typing.List[int]' if sys.version_info >=
```

Comment:
This looks like a symptom of a bug. Can this be fixed?

P.S. In both cases the type is expected to be `JsonArray`. `type='typing.List[int]'` is not a portable type.

Reply:
Actually, I wanted to have `type='typing.List[int]'` instead of just `JsonArray`. 
First, it would make function-based component and yaml-based component symmetric.
Secondly, it carries more info (the inner type) which is necessary for certain scenarios. For example:
```
loop_args: List[int] = ...
with dsl.ParallelFor(loop_args) as item:
   with dsl.Condition(item > value):
       ...
```
To make this condition works in IR, we would need the inner type `int`.